test/expect/TestScript.test_milstm_fusion_cuda-backward.expect

```diff
@@ -20,87 +20,121 @@ graph(%0 : Float(*, *)
-  %34 : Float(*, *) = aten::t(%16)
-  %35 : Float(*, *) = aten::mm(%34, %30)
-  %36 : Float(*, *) = aten::t(%35)
-  %37 : Float(*, *) = aten::t(%17)
-  %38 : Float(*, *) = aten::mm(%37, %32)
-  %39 : Float(*, *) = aten::t(%38)
-  return (%28, %33, %29, %31, %36, %39);
+      %41 : int[]
+      %42 : int[]
+      %43 : int[]
+      %44 : int[]
+      %45 : int[]
+      %46 : int[]
+      %47 : Float(*, *)
+      %48 : int[]
+      %49 : int[]) {
+  %50 : int = prim::Constant[value=1]()
+  %51 : Float(*, *) = aten::mul(%0, %47)
+  %52 : Dynamic = prim::SumToSize(%51, %48)
+  %53 : Float(*, *) = aten::mul(%0, %outgate)
+  %54 : Dynamic = prim::SumToSize(%53, %49)
+  %55 : Dynamic = prim::FusionGroup_0(%1, %54, %47)
+  %56 : Dynamic = prim::SumToSize(%55, %45)
+  %57 : Dynamic = prim::SumToSize(%55, %46)
+  %58 : Dynamic = aten::mul(%57, %cellgate)
+  %59 : Dynamic = prim::SumToSize(%58, %43)
+  %60 : Dynamic = aten::mul(%57, %ingate)
+  %61 : Dynamic = prim::SumToSize(%60, %44)
+  %62 : Dynamic = aten::mul(%56, %12)
+  %63 : Dynamic = prim::SumToSize(%62, %41)
+  %64 : Dynamic = prim::FusionGroup_1(%59, %ingate, %63, %forgetgate, %61, %cellgate, %52, %outgate)
+  %65 : Dynamic = prim::SumToSize(%64, %35)
+  %66 : Dynamic = prim::SumToSize(%64, %36)
+  %67 : Dynamic = prim::SumToSize(%65, %33)
+  %68 : Dynamic = prim::SumToSize(%65, %34)
+  %69 : Dynamic = aten::mul(%68, %Uz)
+  %70 : Dynamic = prim::SumToSize(%69, %31)
+  %71 : Dynamic = aten::mul(%68, %13)
+  %72 : Dynamic = prim::SumToSize(%71, %32)
+  %73 : Dynamic = prim::SumToSize(%67, %29)
+  %74 : Dynamic = prim::SumToSize(%67, %30)
+  %75 : Dynamic = aten::mul(%74, %Wx)
+  %76 : Dynamic = prim::SumToSize(%75, %27)
+  %77 : Dynamic = aten::mul(%74, %14)
+  %78 : Dynamic = prim::SumToSize(%77, %28)
+  %79 : Dynamic = aten::mul(%73, %Uz)
+  %80 : Dynamic = prim::SumToSize(%79, %25)
+  %81 : Dynamic = aten::mul(%73, %24)
+  %82 : Dynamic = prim::SumToSize(%81, %26)
+  %83 : Dynamic = aten::add(%72, %82, %50)
+  %84 : Dynamic = aten::mul(%80, %Wx)
+  %85 : Dynamic = prim::SumToSize(%84, %22)
+  %86 : Dynamic = aten::mul(%80, %15)
+  %87 : Dynamic = prim::SumToSize(%86, %23)
+  %88 : Dynamic = aten::add(%78, %87, %50)
+  %89 : Float(*, *) = aten::t(%16)
+  %90 : Float(*, *) = aten::mm(%89, %83)
+  %91 : Float(*, *) = aten::t(%90)
+  %92 : Float(*, *) = aten::t(%17)
+  %93 : Float(*, *) = aten::mm(%92, %88)
+  %94 : Float(*, *) = aten::t(%93)
+  return (%66, %70, %76, %85, %91, %94);
 }
 with prim::FusionGroup_0 = graph(%0 : Float(*, *)
+      %1 : Dynamic
+      %2 : Float(*, *)) {
+  %3 : Float(*, *) = aten::mul(%2, %2)
+  %4 : Float(*, *) = aten::neg(%3)
+  %5 : int = prim::Constant[value=1]()
+  %6 : Float(*, *) = aten::add(%4, %5, %5)
+  %7 : Dynamic = aten::mul(%1, %6)
+  %8 : int = prim::Constant[value=1]()
+  %9 : Dynamic = aten::add(%0, %7, %8)
+  return (%9);
+}
+with prim::FusionGroup_1 = graph(%0 : Dynamic
```

Comment:
There were three fusion groups before but now it turned into two -- that is interesting

Reply:
Huh, that's surprising! One fusion group in LSTM got split into two instead.